### ISSUE TITLE
Categorical vs. Binnary for many components

### ISSUE THREAD

**Issue by mmelodea** [fchollet/keras#8040]
Hi guys,

I have a question, not sure if here is the right place to ask. I apologize if not.
I'm wondering what is the best solution to work on the following problem: imagine you have a signal class and multiple background classes. Would be better to train a Keras network using categorical classification or using a binnary one (for that one needs to consider all background classes as one merged)? Also, consider that each background class (as well signal) has different amount of examples. So, if one use all background example together would Keras learn less about the classes which have less example than the others?

**Comment by mmelodea** [comment]
Hi guys,

I have a question, not sure if here is the right place to ask. I apologize if not.
I'm wondering what is the best solution to work on the following problem: imagine you have a signal class and multiple background classes. Would be better to train a Keras network using categorical classification or using a binnary one (for that one needs to consider all background classes as one merged)? Also, consider that each background class (as well signal) has different amount of examples. So, if one use all background example together would Keras learn less about the classes which have less example than the others?

**Comment by fchollet** [comment]
It is impossible to know in advance which approach would work better, so you should try both. Having imbalanced classes in an issue during training, which you can handle either by:

- subsample the more common classes, in order to compose batches of training data where the classes are balanced. This means that the network will see the rarer class samples more often.
- use the `class_weight` argument (in `fit`) to increase the loss contribution of rarer classes. This should be done proportionately to their frequency (e.g. if class A has 10k samples and class B has 20k samples, class A should be given weight 2 and class B weight 1.
